docs/CSharp_API.md

```diff
@@ -8,107 +8,207 @@ The Microsoft.ML.OnnxRuntime Nuget package includes the precompiled binaries for
+                 NamedOnnxValue.CreateFromTensor<float>("name2", t2)
+             };
+using (var results = session.Run(inputs))
+{
+    // manipulate the results
+}
+```
+
+You can load your input data into Tensor<T> objects in several ways. A simple example is to create the Tensor from arrays.
+
+```cs
+float[] sourceData;  // assume your data is loaded into a flat float array
+int[] dimensions;    // and the dimensions of the input is stored here
+Tensor<float> t1 = new DenseTensor<float>(sourceData, dimensions);
+```
+
+Here is a [complete sample code](../csharp/sample/Microsoft.ML.OnnxRuntime.InferenceSample) that runs inference on a pretrained model.
+
+## Reuse input/output tensor buffers
+
+In some scenarios, you may want to reuse input/output tensors. This offen happens when you want to chain 2 models (ie. feed one's output as input to another), or want to accelerate inference speed during multiple inference runs.
+
+### Chaining: Feed model A's output(s) as input(s) to model B
+
+```cs
+InferenceSession session1, session2;  // let's say 2 sessions are initialized
+
+Tensor<float> t1;  // let's say data is fed into the Tensor objects
+var inputs1 = new List<NamedOnnxValue>()
+              {
+                  NamedOnnxValue.CreateFromTensor<float>("name1", t1)
+              };
+// session1 inference
+using (var outputs1 = session1.Run(inputs1))
+{
+    // get intermediate value
+    var input2 = outputs1.First();
     
-    Tensor<float> t1, t2;  // let's say data is fed into the Tensor objects
-    var inputs = new List<NamedOnnxValue>()
-                 {
-                    NamedOnnxValue.CreateFromTensor<float>("name1", t1),
-                    NamedOnnxValue.CreateFromTensor<float>("name2", t2)
-                 };
-    using (var results = session.Run(inputs))
+    // modify the name of the ONNX value
+    input2.Name = "name2";
+
+    // create input list for session2
+    var inputs2 = new List<NamedOnnxValue>() { input2 };
+
+    // session2 inference
+    using (var results = session2.Run(inputs2))
     {
         // manipulate the results
     }
+}
+```
 
-You can load your input data into Tensor<T> objects in several ways. A simple example is to create the Tensor from arrays.
+### Multiple inference runs with fixed sized input(s) and output(s)
 
-    float[] sourceData;  // assume your data is loaded into a flat float array
-    int[] dimensions;    // and the dimensions of the input is stored here
-    Tensor<float> t1 = new DenseTensor<float>(sourceData, dimensions);    
+If the model have fixed sized inputs and outputs of numeric tensors, you can use "FixedBufferOnnxValue" to accelerate the inference speed. By using "FixedBufferOnnxValue", the contianer objects only need to be allocated/disposed one time during multiple InferenceSession.Run() calls. This avoids some overhead which may beneficial for smaller models where the time is noticeable in the overall running time.
```

Comment:
Spelling Error "contianer" should be "container".  

Also, "may beneficial" should read "may be beneficial".

Reply:
fixed in latest commit